src/grpc_server.cc

```diff
@@ -3679,6 +3679,14 @@ ModelInferHandler::Process(InferHandler::State* state, bool rpc_ok)
           &state->alloc_payload_ /* response_allocator_userp */,
           InferResponseComplete, reinterpret_cast<void*>(state));
     }
+    // Get request ID for logging in case of error.
+    const char* request_id = nullptr;
+    LOG_TRITONSERVER_ERROR(
+        TRITONSERVER_InferenceRequestId(irequest, &request_id),
+        "unable to retrieve request ID string");
+    if ((request_id != nullptr) && (request_id[0] != '\0')) {
```

Comment:
do we need these checks for null terminator at start / empty string? Was there an edge case?